src/plugins/telemetry/public/plugin.ts

```diff
@@ -321,6 +316,9 @@ export class TelemetryPlugin implements Plugin<TelemetryPluginSetup, TelemetryPl
 
     const currentKibanaVersion = this.currentKibanaVersion;
 
+    const { getTelemetryAllowChangingOptInStatus, getTelemetryOptIn, getTelemetrySendUsageFrom } =
+      await import('../common/telemetry_config');
```

Comment:
Same question here

Reply:
This import only runs when the right APIs are called. I think this does bring some benefits to the initial load. Happy to revert it if you think otherwise.

Reply:
Right sorry, didn't expand the code enough, I thought it was a depth-0 block